src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -711,43 +713,6 @@ var ReactDOMFiberComponent = {
     }
   },
 
-  /**
-   * Destroys all event registrations for workInProgress instance. Does not remove from
-   * the DOM. That must be done by the parent.
-   *
-   * @internal
-   */
-  unmountComponent: function(safely, skipLifecycle) {
-    switch (workInProgress._tag) {
-      case 'input':
-      case 'textarea':
-        inputValueTracking.stopTracking(workInProgress);
-        break;
-      case 'html':
-      case 'head':
-      case 'body':
-        /**
-         * Components like <html> <head> and <body> can't be removed or added
-         * easily in a cross-browser way, however it's valuable to be able to
-         * take advantage of React's reconciliation for styling and <title>
-         * management. So we just document it and throw in dangerous cases.
-         */
-        invariant(
```

Comment:
It is covered by tests so we can get back to it. I'm not sure how to do this since there is no unmount phase for each node. There is only a delete at the top. We could possible check if the tag of every `removeChild` is either of these or if it is html or head then check a few levels deeper.